js/sales.js

```diff
@@ -24,56 +24,119 @@
+                newHourHead.textContent = '12:00am';
+            } else if(militaryHour < 12){
                 //1am-11am
-                newHourItem.textContent = militaryHour + 'am: ' + hourSoldCookies + " cookies";
-            } else if(militaryHour === 12){
-                newHourItem.textContent = '12pm: ' + hourSoldCookies + ' cookies';
-            } else if(militaryHour === 24){
-                newHourItem.textContent = '12am: ' + hourSoldCookies + ' cookies';
+                newHourHead.textContent = militaryHour + ':00am';
             } else {
                 //1pm-11pm
-                newHourItem.textContent = (militaryHour - standardTimeOffSet) + 'pm: ' + hourSoldCookies + ' cookies';
+                newHourHead.textContent = (militaryHour - standardTimeOffSet) + ':00pm';
+            }
+            tableRow.appendChild(newHourHead);
+        }
+        return tableRow;
+    }
+
+    CookieStore.prototype.render = function(){
+        var tableStoreRow = document.createElement('tr');
+        tableStoreRow.setAttribute('scope', 'row');
+        var tableStoreHead = document.createElement('th');
+        tableStoreHead.textContent = this.name;
+        document.querySelector('tbody').appendChild(tableStoreRow);
+        tableStoreRow.appendChild(tableStoreHead);
+
+        calculateSimulatedCookiesPerHour(this);
+        
+        for(var i = 0; i < this.simulatedCookiesPerHour.length; i++){
+            var tableStoreData = document.createElement('td');
+            tableStoreData.textContent = this.simulatedCookiesPerHour[i];
+            tableStoreRow.appendChild(tableStoreData);
+        }
+        return tableStoreRow;
+    }
+
+    function createTotalsFooterRow(){
+        var totalsRow = document.createElement('tr');
+        totalsRow.setAttribute('scope', 'row');
+        var totalsHead = document.createElement('th');
+        totalsHead.textContent = 'Totals';
+        document.querySelector('tfoot').appendChild(totalsRow);
+        totalsRow.appendChild(totalsHead);
+
+        for(var i = 0; i < cookieStoreLocations[0].simulatedCookiesPerHour.length; i++){
+            var total = 0;
+            for(var j = 0; j < cookieStoreLocations.length; j++){
+                total += cookieStoreLocations[j].simulatedCookiesPerHour[i];
             }
-            list.append(newHourItem);
+            var tableTotalData = document.createElement('td');
+            tableTotalData.textContent = total;
+            totalsRow.appendChild(tableTotalData);
+        }
+        return totalsRow;
+    }
+
+    function createStoreTotals(location, tableRow){
+        var total = 0;
+        for(var i = 0; i < location.simulatedCookiesPerHour.length; i++){
+            total += location.simulatedCookiesPerHour[i];
         }
+        var tableStoreTotalData = document.createElement('td');
+        tableStoreTotalData.textContent = total;
+        tableRow.appendChild(tableStoreTotalData);
+        return total;
```

Comment:
While it's tempting to have the logic that appends the total to a row and the logic that keeps a running total of the row in the same function, the problem is that we're creating [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). This function:
- Creates a store total `<td>` element,
- Mutates a `<tr>` element passed in, AND
- Sums the total for a given location, returning that value

So it's not readily apparent to future developers what the purpose of this function is, because it does more than what it's named for.

We could refactor to have something like:
![Capture](https://user-images.githubusercontent.com/20998142/85915111-adde7b00-b7f9-11ea-8dbb-31cd1d496b95.PNG)

In that way, we have a function that generates totals for a location (and really, `calculateTotal` should be a prototype method!), creates and returns `<td>` cells, then a "main" function that uses the return values of these smaller, single-purpose methods.